pymeasure/instruments/ipgphotonics/yar.py

```diff
@@ -0,0 +1,178 @@
+        value = value.upper()
+        if value in values:
+            return value
+    raise ValueError(f"Value is {value}, but a boolean or 'ON' or 'OFF' required.")
+
+
+def setpoint_validator(value, values):
+    if value == 0:
+        return value
+    else:
+        return validators.strict_range(value, values)
+
+
+def power_get_process_generator(minimum):
+    """Generate a get_process for the power property."""
+    def get_process(value):
+        if isinstance(value, float):
+            return value
+        elif value == "Off":
+            return 0
+        elif value == "Low":
+            return minimum
+        else:
+            return value
+    return get_process
+
+
+class YAR(Instrument):
+    """Communication with the YAR fiber amplifier series by IPG Photonics.
+
+    This is the RS232 command set. GPIB has different commands.
+    """
+
+    def __init__(self, adapter, name="YAR fiber amplifier", **kwargs):
+        """Establish communication with the device."""
+        kwargs.setdefault("write_termination", "\r")
+        kwargs.setdefault("read_termination", "\r")
+        super().__init__(adapter,
+                         name=name,
+                         includeSCPI=False,
+                         asrl={'parity': Parity.none, 'stop_bits': StopBits.one},
+                         **kwargs)
+
+        # Commands are 3-4 letters, followed by a parameter, a separation
+        # by space is option. Commands are case-insensitive.
+        # Response is command echoed back, followed by ': ' and the return
+        # value.
+
+        # get valid range of power setpoint:
+        self.power_setpoint_values = self.power_range
+        self.power_get_process = power_get_process_generator(self.minimum_display_power)
+
+    class Status(IntFlag):
+        EMISSION = 0x1  # emission is fully on
+        STARTUP_DELAY = 0x2  # it is in 3 s startup
+        HIGH_TEMPERATURE = 1 << 16
+        HIGH_BACKREFLECTION = 1 << 17
+        UNEXPECTED_EMISSION = 1 << 19
+        SEEDLASER_FAIL = 1 << 20
+
+    def read(self):
+        """Read an instrument answer and check whether it is an error."""
+        reply = super().read().split(":")
+        if reply[0] == "ERR":
+            raise ConnectionError(f"Reading error {reply}")
+        else:
+            return reply[-1].strip()
+
+    def check_errors(self):
+        """Empty the read buffer."""
+        got = self.read().split(":")
+        if got[0] == "ERR":
+            raise ConnectionError(f"Connection error {got[-1]}.")
+
+    # COMMUNICATION FUNCTIONS
+
+    @property
+    def id(self):
+        """Get the model number."""
+        return self.values("RMN")[0]
+
+    @property
+    def status(self):
+        """Get the current status."""
+        got = int(self.values("STA")[0])
+        return self.Status(got)
+
+    emission = Instrument.control(
+        "STA",
+        "EM%s",
+        """Control emission of the amplifier (bool).""",
```

Comment:
In line with our guidelines, I think `emission_enabled` might be a better name for this boolean property.

Reply:
Ok, done.